### PR TITLE
DO NOT MERGE - DDE filtering

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
@PhilipRosedale Ready for test and review.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d60ee3b6e24590d35c3233338261eec0b7954ac9 available at: https://gist.github.com/48271230cf4beb115570<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.dmg

**Comment by Atlante45** [comment]
Code looks good, but I am concerned that non Menu related code made it's way back to the menu class.

It feels to me like those methods could easily live in there related objects:
```
// Move to Discoverability manager
void setVisibility();
void visibilityChanged(Discoverability::Mode discoverabilityMode); 
// Move to Application
void setActiveFaceTracker();
```

This does not block this PR and can be done in a later one, but if you wouldn't mind moving it, that'd be great.

**Comment by cozza13** [comment]
I just tested this and it works great.

**Comment by cozza13** [comment]
Hi @ctrlaltdavid this works, can you remove "DO Not Merge" part of the comment and I will get merged

**Comment by ctrlaltdavid** [comment]
It is crashing for Philip on OSX, hence "DO NOT MERGE".

**Comment by Atlante45** [comment]
Hmmm interesting, I played with it quite a bit to see the improvement versus no filtering and I haven't crashed yet.
@PhilipRosedale did you get a stacktrace?

**Comment by ctrlaltdavid** [comment]
@Atlante45 I've e-mailed you Philip's stack trace. If you can't see anything and it's working OK for you and Chris, can probably merge.

File: interface/src/devices/DdeFaceTracker.cpp [hunk 1]
```diff
@@ -341,11 +380,19 @@ void DdeFaceTracker::decodePacket(const QByteArray& buffer) {
...
+        // Calculate average frame time
+        const float FRAME_AVERAGING_FACTOR = 0.99f;
+        quint64 usecsNow = usecTimestampNow();
+        if (_lastMessageReceived != 0) {
+            _averageMessageTime = FRAME_AVERAGING_FACTOR * _averageMessageTime 
+                + (1.0f - FRAME_AVERAGING_FACTOR) * (float)(usecsNow - _lastMessageReceived) / 1000000.0f;
+        }
+        _lastMessageReceived = usecsNow;
+    
+} else {
```
Comment (Atlante45) [inline interface/src/devices/DdeFaceTracker.cpp]: indentation

**Comment by Atlante45** [comment]
@ctrlaltdavid yup thanks, I'm looking at it right now.
This is definitely avatar related, but I think we should hold up on merge until we figure out what's going on.
If 1 out of 3 of our mac users crash on startup, that's no bueno.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b085ca6684887511b447cd24b0dea059784aa3c available at: https://gist.github.com/48271230cf4beb115570<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.dmg

**Comment by cozza13** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b085ca6684887511b447cd24b0dea059784aa3c available at: https://gist.github.com/48271230cf4beb115570<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee480248315e07c91db495674409313f6a446b94 available at: https://gist.github.com/48271230cf4beb115570<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4658.dmg

**Pull request merged**